dipy/tracking/streamlinespeed.pyx

```diff
@@ -301,3 +301,188 @@ def set_number_of_points(streamlines, nb_points=3):
+        for d in range(D):
+            dn = streamline[i, d] - streamline[last_added_point, d]
+            in_between_dist += dn*dn
+
+        in_between_dist = sqrt(in_between_dist)
+
+        if i == 0:  # First point is always added.
+            for d in range(D):
+                out[nb_points, d] = streamline[i, d]
+
+            last_success = i
+            nb_points += 1
+        elif i < N-1:  # Perform linearization if needed.
+            prev = last_added_point
+            next = i+1
+
+            # Check that each point is not offset by more than `tol_error` mm.
+            for k in range(last_added_point, i):
+                curr = k
+                dist = c_dist_to_line(streamline, prev, next, curr)
+
+                # If the current point's offset is greater than `tol_error`, use the latest success.
+                if dist > tol_error or in_between_dist > max_segment_length:
+                    for d in range(D):
+                        out[nb_points, d] = streamline[last_success, d]
+
+                    nb_points += 1
+                    last_added_point = i-1
+                    last_success = i-1
+                    break
+                # The current point's offset is ok, check the next point.
+                else:
+                    last_success = i
+
+        else:  # Last point is always added.
+            for d in range(D):
+                out[nb_points, d] = streamline[i, d]
+
+            nb_points += 1
+
+    return nb_points
+
+
+def compress_streamlines(streamlines, tol_error=0.1, max_segment_length=10):
+    """ Compress streamlines by linearization as in [Presseau15]_.
+
+    The compression consists in merging consecutive segments that are
+    near collinear. The merging is achieved by removing the point the two
+    segments have in common.
+
+    The linearization process [Presseau15]_ ensures that every points being
+    removed are within a certain margin (in mm) of the resulting streamline.
+    Recommendations for setting this margin can be found in [Presseau15]_
+    (in which they called it tolerance error).
+
+    The compression also ensures that two consecutive points won't be to far
+    from each other (precisely less or equal than `max_segment_length`mm).
+    This is purely to speed up the linearization process [Rheault15]_. A low
+    value will result in a faster linearization but low compression, whereas
+    a high value will result in a slower linearization but high compression.
+
+    Parameters
+    ----------
+    streamlines : one or a list of array-like of shape (N,3)
+        Array representing x,y,z of N points in a streamline
+    tol_error : float (optional)
+        Tolerance error in mm (default: 0.1). A rule of thumb is to set it
+        to 0.01mm for deterministic streamlines and 0.1mm for probabilitic
+        streamlines.
+    max_segment_length : float (optional)
+        Maximum length of any given segment produced by the compression.
```

Comment:
oh wait, that's the max allowed between two points in mm? Then my previous comment is useless. Anyway, you could also add as a note to not do stat on compressed fibers or the result won't be valid at all, just in case people start compressing like crazy and they don't check back changed docstrings.